src/DynamoPackages/PackageLoader.cs

```diff
@@ -342,5 +342,25 @@ internal void DoCachedPackageUninstalls(IPreferences preferences)
 
             preferences.PackageDirectoriesToUninstall.RemoveAll(pkgDirsRemoved.Contains);
         }
+
+        internal void Dispose()
+        {
+            if (RequestLoadNodeLibrary != null)
+            {
+                foreach (var d in RequestLoadNodeLibrary.GetInvocationList())
+                {
+                    RequestLoadNodeLibrary -= (d as Action<Assembly>);
+                }
+            }
+
+            if (RequestLoadCustomNodeDirectory != null)
+            {
+                foreach (var d in RequestLoadCustomNodeDirectory.GetInvocationList())
+                {
+                    RequestLoadCustomNodeDirectory -=
+                        (d as Func<string, IEnumerable<CustomNodeInfo>>);
+                }
+            }
+        }
```

Comment:
Instead of doing it this way, let's dispose of these event handlers in `PackageManagerExtension.Dispose`.